spec/models/lead_provider_api_token_spec.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe LeadProviderApiToken, type: :model do
+  it "generates a hashed token that can be used" do
+    unhashed_token = LeadProviderApiToken.create_with_random_token!(lead_provider: create(:lead_provider))
+
+    expect(
+      LeadProviderApiToken.find_by_unhashed_token(unhashed_token),
+    ).to eql(LeadProviderApiToken.last)
```

Comment:
We should have an `order(:created_at)` here

Reply:
Actioned here c2f792f5